actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -114,18 +130,31 @@ def _compute_redirect_to_location(request, options) #:nodoc:
     public :_compute_redirect_to_location
 
     private
-      def _extract_redirect_to_status(options, response_status)
+      def _compute_safe_redirect_to_location(request, options, response_options)
+        location = _compute_redirect_to_location(request, options)
+        location_options = options.is_a?(Hash) ? options : {}
+        if response_options[:allow_other_host] || _url_host_allowed?(location, location_options)
+          location
+        else
+          fallback_location = response_options.fetch(:fallback_location) do
+            raise RedirectFallbackError, location
```

Comment:
Unsure about the use of a custom error class for fallbacks. We should probably make `redirect_back` throw the same error for consistency too if we choose to use it. Would appreciate some feedback on this.

Reply:
I'd keep the an argument error like it is with `redirect_back`